scripts/validate_docstrings.py

```diff
@@ -471,6 +471,36 @@ def parameter_desc(self, param):
                 desc = desc[:desc.index(full_directive)]
         return desc
 
+    def parameter_desc_list(self, param):
+        i = list(self.doc_parameters).index(param)
+        desc_list = [line
+                     for line in self.doc._parsed_data['Parameters'][i][-1]
+                     if line]
+        # Find and strip out any sphinx directives
+        for directive in DIRECTIVES:
```

Comment:
Do these not cause a problem with non-bulleted descriptions? Seems like a test needs to hit this somewhere

Reply:
Hi @WillAyd ,

What exactly are you referring to?

If you are referring to the part that "Find and strip out any sphinx directives"
I have replicated the check that was already present in the method `parameter_desc`

Reply:
My point was I don't see the purpose of this code nor a test that hits it

Reply:
I am not sure why the code was added in the method `parameter_desc`. I wanted to replicate the functionality in `parameter_desc_list` for consistency so the two methods return the same parameter description text either as a unique string (`parameter_desc`) or as a list of strings (`parameter_desc_list`).

It seems that the description text after some directives ('versionadded', 'versionchanged', 'deprecated') should be ignored by the validator.

I am adding some tests for docstrings that contain both bullet points and directives (similar to GoodDocStrings.mode and BadParameters.no_description_period_with_directive).

Reply:
Hi @WillAyd ,
I have added the tests. Please let me know if I should make other changes.

Reply:
Thanks @EdAbati . I'm limited on availability until Tuesday will take a look next week